crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -108,3 +108,105 @@ fn create_directories() -> Result<(), anyhow::Error> {
     )?;
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use c8y_api::http_proxy::MockC8yJwtTokenRetriever;
+    use c8y_smartrest::smartrest_deserializer::SmartRestJwtResponse;
+    use mockito::mock;
+    use mqtt_tests::{assert_received_all_expected, test_mqtt_broker};
+    use serde_json::json;
+    use std::time::Duration;
+    use test_case::test_case;
+
+    const TEST_TIMEOUT_SECS: Duration = Duration::from_secs(5);
+    const CUMULOCITY_MAPPER_NAME_TEST: &str = "tedge-mapper-c8y-test";
+    const DEVICE_ID: &str = "test-device";
+    const DEVICE_NAME: &str = "test-user";
+    const DEVICE_TYPE: &str = "test-thin-edge.io";
+    const MQTT_HOST: &str = "127.0.0.1";
+
+    #[test_case(
+        "tedge/measurements",
+        "tedge/errors",
+        "{",
+        "Invalid JSON: EOF while parsing an object at line 1 column 1: `{`"
+    )] // fail case
+    #[test_case(
+        "tedge/measurements",
+        "tedge/measurements",
+        r#"{"temperature": 12, "time": "2021-06-15T17:01:15.806181503+02:00"}"#,
+        r#"{"temperature": 12, "time": "2021-06-15T17:01:15.806181503+02:00"}"#
+    )] // successs case
+    #[tokio::test]
+    async fn test_tedge_mapper_with_mqtt_pub(
+        pub_topic: &str,
+        sub_topic: &str,
+        payload: &str,
+        expected_msg: &str,
+    ) -> Result<(), anyhow::Error> {
+        // Mock endpoint to return C8Y internal id
+        let _get_internal_id_mock = mock("GET", "/identity/externalIds/c8y_Serial/test-device")
+            .with_status(200)
+            .with_body(
+                json!({ "externalId": DEVICE_ID, "managedObject": { "id": "123" } }).to_string(),
+            )
+            .create();
+
+        let mut jwt_token_retriver = Box::new(MockC8yJwtTokenRetriever::new());
+        jwt_token_retriver
+            .expect_get_jwt_token()
+            .returning(|| Ok(SmartRestJwtResponse::default()));
+
+        let http_client = reqwest::ClientBuilder::new().build().unwrap();
```

Comment:
I see only http here, is it using https in the background? I'm just curious how we could manage and test https  requests.

Reply:
You are correct, the tests are done using http not https. An issue like [this one](https://github.com/thin-edge/thin-edge.io/issues/1082) will not be detected by these tests.